File: BLEtest/app/src/main/java/com/example/bletest/MainActivity.java
package com.example.bletest;

import androidx.activity.result.ActivityResultLauncher;
import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.os.Build;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //check android12+
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            ActivityResultLauncher<I> requestMultiplePermissions;
            requestMultiplePermissions.launch(arrayOf(
                    Manifest.permission.BLUETOOTH_SCAN,
                    Manifest.permission.BLUETOOTH_CONNECT))
        }
        else{
            val enableBtIntent = Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE)
            requestBluetooth.launch(enableBtIntent)
        }
....................................................

        private var requestBluetooth = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
            if (result.resultCode == RESULT_OK) {
                //granted
            }else{
                //deny
            }
        }

        private val requestMultiplePermissions =
                registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
                permissions.entries.forEach {
            Log.d("test006", "${it.key} = ${it.value}")
        }
        }
    }
}

File: BLE/app/src/main/java/com/example/ble/MainActivity.kt
package com.example.ble

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var binding:ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding=A
        setContentView(R.layout.activity_main)



//
//        var requestBluetooth = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
//            if (result.resultCode == RESULT_OK) {
//                //granted
//                showToast("granted")
//            }else{
//                //deny
//                showToast("deny")
//            }
//        }
//
//        val requestMultiplePermissions =
//            registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
//                permissions.entries.forEach {
//                    Log.d("test006", "${it.key} = ${it.value}")
//                }
//            }
//        showToast(requestBluetooth.toString())
    }

    // Receiver
    //функция всплывающего сообщения
    private fun showToast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }
}